src/renderers/webgl/WebGLPrograms.js

```diff
@@ -163,7 +163,7 @@ function WebGLPrograms( renderer, cubemaps, cubeuvmaps, extensions, capabilities
 			isWebGL2: IS_WEBGL2,
 
 			shaderID: shaderID,
-			shaderName: material.type,
+			shaderName: ( ( material.isShaderMaterial || material.isRawShaderMaterial ) && material.name !== '' ) ? material.name : material.type,
```

Comment:
What about...

```js
shaderName: `${ material.type } (${ material.name })`,
```

Reply:
1. IMO,  "ShaderMaterial FilmShader" is not necessary. "FilmShader" is sufficient. (see screenshot at top ).
2. It's not appropriate for Built-in materials. Each instance of `MeshBasicMaterial` shares the same shader.

Reply:
The code would be simpler and the behavior would be consistent for all materials.

I can't see what the downside is?

Reply:
Because a user can do this:
```js
const m1 = new THREE.MeshBasicMaterial();
const m2 = new THREE.MeshBasicMaterial();
m1.name = 'foo';
m2.name = 'bar';
```

Then it will list the material once as 'MeshBasicMaterial foo' because they share the same shader.

Is that OK with you? I think that would be confusing...

Reply:
Also note, with the concatenated name as you suggest, the shader code will be
```glsl
define SHADER_NAME MeshBasicMaterial foo
```

Reply:
@mrdoob @Mugen87 I'll implement whatever you both agree to... Just pointing out my concerns.

Reply:
How about using two defines? `SHADER_TYPE` and `SHADER_NAME`. 

`SHADER_TYPE` would match to `Material.type` and `SHADER_NAME` to `Material.name`.

Reply:
I think it could be useful for debugging when the name of built-in materials is also accessible in the shader.